torch/_dynamo/decorators.py

```diff
@@ -236,6 +236,24 @@ def mark_static(t, index=None):
             mark_static(t, i)
 
 
+@forbid_in_graph
+def mark_static_address(t, guard=True):
+    """
+    Marks an input tensor whose data_ptr will not change across multiple calls
+    to a dynamo-compiled function. This indicates to cudagraphs that an extra allocation
+    is not needed for this input. The data_ptr will be guarded if guard=True. Note:
+    Until https://github.com/pytorch/pytorch/issues/104095 is resolved, tensors marked
```

Comment:
Even if https://github.com/pytorch/pytorch/issues/104095, is partially fixed with the ID_Match thing weakref thing, the tensors would be kept alive if this invoked on a free function. 

And we dont have any particular proposal of how clean-up should occur with free functions.

Maybe better to just say these are kept alive ?

Reply:
Yeah I can do that, I'll likely have to do the finalizer approach for the optimizer